google/cloud/pubsub/samples/subscription_admin_samples.cc

```diff
@@ -52,6 +54,33 @@ CreateSubscriptionAdminCommand(std::string const& name,
                                                            std::move(adapter)};
 }
 
+// Delete all subscriptions created with that include "cloud-cpp-samples".
+// Ignore any failures. If multiple tests are cleaning up subscriptions in
+// parallel, then the delete call might fail.
+void CleanupSubscriptions(
+    google::cloud::pubsub_admin::SubscriptionAdminClient& client,
+    std::string const& project_id, absl::Time time_now) {
+  for (auto const& subscription : client.ListSubscriptions(
+           google::cloud::Project(project_id).FullName())) {
+    if (!subscription) continue;
+    std::string const keyword = "cloud-cpp-samples";
+    if (!absl::StrContains(subscription->name(), keyword)) continue;
+    // Extract the date from the resource name which is in the format
+    // `*-cloud-cpp-samples-YYYY-MM-DD-`.
+    auto date = subscription->name().substr(
+        subscription->name().find(keyword) + keyword.size() + 1, 10);
```

Comment:
looking up the substring twice is odd. Also, it took me a second to realize the `+ 1` was the hyphen. We can just encode that in the keyword.

```
    std::string const keyword = "cloud-cpp-samples-";
    auto pos = subscription->name().find(keyword);
    if (pos == std::string::npos) continue;
    // Extract the date from the resource name which is in the format
    // `*-cloud-cpp-samples-YYYY-MM-DD-`.
    auto date = subscription->name().substr(pos + keyword.size(), 10);
```